src/tools/crawling_tools.py

```
@@ -991,34 +1020,201 @@ async def _process_code_examples(ctx: Context, crawl_results: List[Any], directo
+    
+    if file_ext in ['.py', '.js', '.ts', '.java', '.cpp', '.c', '.go', '.rs']:
+        return "SourceCode"
+    elif file_ext in ['.md', '.rst', '.txt']:
+        return "Documentation"
+    elif file_ext in ['.json', '.yaml', '.toml', '.ini']:
+        return "Configuration"
+    else:
+        return "TextFile"
+
+
+def _create_file_node(result, directory_path: str, file_type: str, sections: list, code_blocks: list, mentioned_apis: list):
```

Ruff (0.12.2)

Missing return type annotation for private function `_create_file_node`

(ANN202)